tests/s25Main/integration/testGameServerJointPlayerDistribution.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2016 - 2018 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#include "BasePlayerInfo.h"
+#include "JoinPlayerInfo.h"
+#include "network/GameServer.h"
+#include "gameTypes/TeamTypes.h"
+#include <boost/test/tools/old/interface.hpp>
+#include <boost/test/unit_test.hpp>
+#include <cstdio>
+
+BOOST_AUTO_TEST_SUITE(GameServerPlayerSuite)
+
+static unsigned getMaxTeamSizeDifference(std::vector<JoinPlayerInfo>& playerInfos, unsigned nTeams)
+{
+    unsigned nPlayers[4] = {0, 0, 0, 0};
+    for(auto& playerInfo : playerInfos)
+    {
+        switch(playerInfo.team)
+        {
+            case TM_RANDOMTEAM:
+            case TM_TEAM1: ++nPlayers[0]; break;
+            case TM_RANDOMTEAM2:
+            case TM_TEAM2: ++nPlayers[1]; break;
+            case TM_RANDOMTEAM3:
+            case TM_TEAM3: ++nPlayers[2]; break;
+            case TM_RANDOMTEAM4:
+            case TM_TEAM4: ++nPlayers[3]; break;
+            default: return -1;
+        };
+    }
+
+    std::sort(&nPlayers[0], &nPlayers[nTeams]);
+    BOOST_REQUIRE_LE(nPlayers[0], nPlayers[nTeams - 1]);
```

Comment:
I found it easier to switch to the new Boost test macros:
```suggestion
    BOOST_TEST_REQUIRE(nPlayers[0] < nPlayers[nTeams - 1]);
```